lib/web/apiserver.go

```diff
@@ -282,6 +283,30 @@ func NewHandler(cfg Config, opts ...HandlerOption) (*RewritingHandler, error) {
 			return
 		}
 
+		// check proxy and auth node versions (breaking change from v5).
+		// If there is a mismatch with the current proxy client, redirect users to an error page to update proxy.
+		//
+		// DELETE IN 5.0: this block only serves to notify users of v4.3+ to upgrade to v5.0.0
+		if strings.HasPrefix(r.URL.Path, "/web/newuser") {
+			res, err := h.auth.Ping(context.TODO())
+			if err != nil {
+				log.WithError(err).Debugf("Could not ping auth server")
+			} else {
+				proxyVersion := strings.Index(teleport.Version, "4.")
+				authVersion := strings.Index(res.GetServerVersion(), "5.")
+
+				if authVersion == 0 && proxyVersion == 0 {
+					message := "There is a version mismatch between your proxy and auth services. Please upgrade your proxy service to version 5.0.0 or higher."
```

Comment:
> There is a version mismatch between your proxy and auth services. Please upgrade your proxy service to version 5.0.0 or higher."

@benarent  What do you think about this message instead? -> "Your Teleport proxy service version is not compatible with auth. service. Please update your Teleport proxy service to [place auth. version here]"

Reply:
I think having a rolling version of the auth service is a good idea.  I would stick with upgrading, as we use this in our customer facing docs.

Reply:
Revised to `"Your Teleport proxy and auth service versions are incompatible. Please upgrade your Teleport proxy service to version %v"` 

commit https://github.com/gravitational/teleport/pull/3462/commits/9a3e5f265f60905fbbb1ae8a7e49de4b86cf1ba9